octopod-backend/src/Types.hs

```diff
@@ -19,25 +19,20 @@ module Types
   )
 where
 
-import Data.Bifunctor
 import Data.ByteString (ByteString)
+import qualified Data.ByteString.Lazy as BS
+import Data.Csv
+import Data.Foldable
 import Data.Text as T
-import Data.Traversable
+import Data.Text.Encoding as T
 
 import Common.Types
 
 -- | Parses deployment metadata.
-parseDeploymentMetadata :: [Text] -> IO [DeploymentMetadata]
-parseDeploymentMetadata texts =
-  for texts $ \t ->
-    case T.findIndex (== ',') t of
-      Just i -> do
-        let (key, value) = bimap strip (T.tail . strip) $ T.splitAt i t
-        pure $ DeploymentMetadata key value
-      Nothing ->
-        error $
-          "Malformed metadata key-value pair " <> T.unpack t
-            <> ", should be similar to foo,bar"
+parseDeploymentMetadata :: Text -> IO [DeploymentMetadata]
+parseDeploymentMetadata text = case decode NoHeader $ BS.fromStrict $ T.encodeUtf8 text of
+  Left err -> error $ "Could not parse metadata: " <> err
```

Comment:
1. [ ] We really shouldn't have `error` anywhere in server code.
   
   This should be properly handled. (log an error, return an error from the Servant handler if it is in a request.
   
2. [ ] There is no reason for this to be `IO`.